api/src/test/java/com/epam/pipeline/acl/cluster/ClusterApiServiceTest.java

```diff
@@ -0,0 +1,432 @@
+        doReturn(authentication).when(mockAuthManager).getAuthentication();
+        initAclEntity(nodeInstance, AclPermission.READ);
+        initAclEntity(anotherNodeInstance, AclPermission.NO_READ);
+        doReturn(mutableListOf(nodeInstance)).when(mockNodesManager).filterNodes(filterNodesVO);
+
+        assertThat(clusterApiService.filterNodes(filterNodesVO)).hasSize(1).contains(nodeInstance);
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldReturnEmptyFilteredNodeInstanceListWhenPermissionIsNotGranted() {
+        initAclEntity(anotherNodeInstance);
+        doReturn(mutableListOf(nodeInstance)).when(mockNodesManager).filterNodes(filterNodesVO);
+
+        assertThat(clusterApiService.filterNodes(filterNodesVO)).isEmpty();
+    }
+
+    @Test
+    @WithMockUser(roles = ADMIN_ROLE)
+    public void shouldReturnNodeInstanceForAdmin() {
+        doReturn(nodeInstance).when(mockNodesManager).getNode(nodeInstance.getName());
+
+        assertThat(clusterApiService.getNode(nodeInstance.getName())).isEqualTo(nodeInstance);
+    }
+
+    @Test
+    @WithMockUser(username = SIMPLE_USER)
+    public void shouldReturnNodeInstanceWhenPermissionIsGranted() {
+        final NodeInstance nodeInstance = NodeCreatorUtils.getNodeInstance(1L, TEST_STRING);
+        initAclEntity(nodeInstance, AclPermission.READ);
+        doReturn(SIMPLE_USER).when(mockAuthManager).getAuthorizedUser();
+        doReturn(nodeInstance).when(mockNodesManager).getNode(nodeInstance.getName());
+        doReturn(pipelineRun).when(mockPipelineRunManager).loadPipelineRun(eq(pipelineRun.getId()));
+
+        final NodeInstance returnedNodeInstance = clusterApiService.getNode(nodeInstance.getName());
+
+        assertThat(returnedNodeInstance).isEqualTo(nodeInstance);
+        assertThat(returnedNodeInstance.getMask()).isEqualTo(AclPermission.READ.getMask());
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldDenyAccessToNodeWhenPermissionIsNotGranted() {
+        initAclEntity(nodeInstance);
+        doReturn(nodeInstance).when(mockNodesManager).getNode(nodeInstance.getName());
+        doReturn(pipelineRun).when(mockPipelineRunManager).loadPipelineRun(eq(pipelineRun.getId()));
+
+        assertThrows(AccessDeniedException.class, () -> clusterApiService.getNode(nodeInstance.getName()));
+    }
+
+    @Test
+    @WithMockUser(roles = ADMIN_ROLE)
+    public void shouldReturnNodeThroughRequestForAdmin() {
+        doReturn(nodeInstance).when(mockNodesManager).getNode(nodeInstance.getName(), filterPodsRequest);
+
+        assertThat(clusterApiService.getNode(nodeInstance.getName(), filterPodsRequest)).isEqualTo(nodeInstance);
+    }
+
+    @Test
+    @WithMockUser(username = SIMPLE_USER)
+    public void shouldReturnNodeThroughRequestWhenPermissionIsGranted() {
+        final NodeInstance nodeInstance = NodeCreatorUtils.getNodeInstance(1L, TEST_STRING);
+        initAclEntity(nodeInstance, AclPermission.READ);
+        initMocksBehavior(true);
```

Comment:
Probably this mocking call makes thing even more complicated then before. For example in this case `nodeInstance` used for acl is not the same instance that will be used in `initMocksBehavior`. Also this `true\false` flag is not intuitive. Just imagine that there will be another boolean parameter.

I still think that it will be better to use separate calls for mocks which will just remove a need to write the whole
`doReturn().when().call()` chain. For example in this case we can use something like the following. It does not contain less lines then the original version. But it is not the point, isn't it? The following code is declarative as plain mockito calls but it is also much simpler. And in case f.e. pipeline run mocking mechanism will have to change in the future then we will just have to change a single method `mockRun`.
```java
mockUser(SIMPLE_USER); // Mock any user related calls that are used in the tests.
mockNode(nodeInstance); // Mock all node instance calls that are used in the tests.
mockRun(pipelineRun); // Mock all pipeline run calls that are used in the tests.
```

```java
public void mockNode(final NodeInstance nodeInstance) {
    doReturn(nodeInstance).when(mockNodesManager).getNode(nodeInstance.getName(), filterPodsRequest);
    doReturn(nodeInstance).when(mockNodesManager).getNode(nodeInstance.getName());
}
```

What do you think?